modules/packages/npm/creator_test.go

```diff
@@ -67,6 +67,16 @@ func TestParsePackage(t *testing.T) {
 		test(t, " test")
 		test(t, "test ")
 		test(t, "te st")
+		test(t, "Test")
+		test(t, "_test")
+		test(t, ".test")
+		test(t, "^test")
+		test(t, "te^st")
+		test(t, "te|st")
+		test(t, "te)(st")
+		test(t, "te'st")
+		test(t, "te!st")
+		test(t, "te*st")
 		test(t, "invalid/scope")
 		test(t, "@invalid/_name")
 		test(t, "@invalid/.name")
```

Comment:
I think we should also test for
> package name cannot be the same as a node.js/io.js core module nor a reserved/blacklisted name. For example, the following names are invalid:
    http
    stream
    node_modules
    favicon.ico

Reply:
Could keep a map of reserved names. Would be cleaner then extending the regex.

Reply:
One doesn't exclude the other?
The test method does not test the regex directly, but the upload.

Reply:
Can we merge this PR without reserved names inspection feature?
Can I create separated issue for that, and I think, I'm able to implement it, but need some time to dive into codebase.